File: app/src/main/java/com/example/resapp/SelectDateActivity.java
package com.example.resapp;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.CalendarView;

import androidx.appcompat.app.AppCompatActivity;

public class SelectDateActivity extends AppCompatActivity {
    private String loggedInUserEmail;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_select_date);
        loggedInUserEmail = getIntent().getStringExtra("user_email");

        CalendarView calendarView = findViewById(R.id.calendarView);
        Button btnManage = findViewById(R.id.btnManage);

        btnManage.setOnClickListener(v -> {
            Intent intent = new Intent(SelectDateActivity.this, ManageReservationActivity.class);
            intent.putExtra("user_email", loggedInUserEmail);
            startActivity(intent);

        });

        calendarView.setOnDateChangeListener((view, year, month, dayOfMonth) -> {
            String selectedDate = dayOfMonth + "/" + (month + 1) + "/" + year;

            Intent intent = new Intent(SelectDateActivity.this, SelectTimeActivity.class);
            intent.putExtra("selectedDate", selectedDate);
            intent.putExtra("user_email", loggedInUserEmail);
            startActivity(intent);
        });
    }
}

File: app/src/main/java/com/example/resapp/SelectDateActivity.kt
package com.example.resapp

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.CalendarView
import androidx.appcompat.app.AppCompatActivity

class SelectDateActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_select_date)

        val calendarView = findViewById<CalendarView>(R.id.calendarView)
        val btnManage = findViewById<Button>(R.id.btnManage)

        btnManage.setOnClickListener {
            val intent = Intent(this, ManageReservationActivity::class.java)
            startActivity(intent)
        }

        calendarView.setOnDateChangeListener { _, year, month, dayOfMonth ->
            val selectedDate = "$dayOfMonth/${month + 1}/$year"

            val intent = Intent(this, SelectTimeActivity::class.java)
            intent.putExtra("selectedDate", selectedDate)
            startActivity(intent)
        }
    }
}
